### PR TITLE
Bagelstation stuff and adds the Barber as a Bagelstation only Job [NEEDS REVIEW]

### LINKED ISSUE #15001: Bagel vox outpost
There's a dead APC in the hallway near engineering

There are no lights in the antechamber to the tarjan genetics room.

There was also an air alarm with no power somewhere.

### LINKED ISSUE #15009: Bagel: engineering oversights
These meters could/should be assigned a name and frequency
dreamseeker_2017-05-24_11-42-56 dreamseeker_2017-05-24_11-42-30

The SME computer is not linked to the injectors in the chamber, I feel like it should be?
dreamseeker_2017-05-24_11-43-42

### LINKED ISSUE #15041: [BAGEL] another reversed intercomm
(WEB REPORT BY: sonixapache REMOTE: 172.93.109.202:7777)
>Revision (Should be above if you're viewing this from ingame!)

3b727e7bfb1e89d66fcc4faa450118ffc43f6684 

>General description of the issue

https://puu.sh/w1Xiu/c5f80eb126.png

### LINKED ISSUE #15052: Bagel: Paramedic APC placement
How did no one report this already?
image

### LINKED ISSUE #15058: missing pipe in bagel atmos
missing pipe in bagel atmospherics

### PR DESCRIPTION
**Pull request by CptWad**
If you were not already aware Barber stuff was added by @Arthurdentist 
This here formalizes it as an official job on Bagelstation.
Basically, this job has zero openings, except on bagelstation where it has one.
Gets Maint and Barbers access. (Only for the vending machine which only he can use) access.dm was set up right i'm pretty sure.

As for tools to cut people's hair, ive gutted the electric razors griffy use for a more utility one, you now should be able to choose the hairstyle you want to give someone (0% tested).

I also cut down on how much stuff was in the barber vend since before it was a bit much.

Starts in the dorms area, in his little barber cubby.
image

Other than that more bagel stuff, fixed kitchen delivery going to janitors, added names and frequencies to some pipes and meters in atmos. Fixed issues on the trader outpost. 
closes #15001
close #15009
closed #15041
fixed #15052
fixes #15058
solves #15105

:cl:
- experimental: Adds barber role, available only to Bagelstation
- tweak: Electric razors now let you choose what to shave their hair into. Floor length Braids for everyone!
- tweak: More bagel bugfixes and microchanges.
- tweak: Now there can be two Wardens on Bagel. One at start and another late-join.

### CODE REVIEW

**Comment by Shadowmech88** [comment]
The existence of mirrors and ordering hair dye from Cargo makes this job entirely useless.

I don't think it's a good idea to add any more RP jobs than we already have. I foresee this job becoming just "assistant but with his own personal room."

**Comment by CptWad** [comment]
I mean I could make two forms of mirror like mirror/nocut which cant cut hair and place those only on bagel. I can also disable said crate on bagel as well no?

**Comment by Arthurdentist** [comment]
@CptWad uniforms came in fours so you could start a quartet

File: code/game/machinery/vending.dm [hunk 1]
```diff
@@ -2447,19 +2447,20 @@ var/global/num_vending_terminals = 1
 /obj/machinery/vending/barber
 	name = "BarberVend"
 	desc = "The ultimate vendor for any aspiring space stylist."
-	product_slogans = "Haircuts for everyone!;Choose your own style!;A new look avaliable now!"
+	req_access = list(access_barber)
+	product_slogans = "Haircuts for everyone!;Choose your own style!;A new look avaliable now!;A cut above!"
```
Comment (Intigracy) [inline code/game/machinery/vending.dm]: avaliable

Comment (CptWad) [inline code/game/machinery/vending.dm]: what?
image

Comment (Exxion) [reply code/game/machinery/vending.dm]: Yes
It should be "available"

File: code/game/objects/items/weapons/cosmetics.dm [hunk 2]
```diff
@@ -310,16 +319,16 @@
 									 "<span class='notice'>You start shaving [H]'s facial hair...</span>")
 				if(do_after(user, H, 50))
 					user.visible_message("<span class='warning'>[user] shaves off [H]'s facial hair with [src].</span>", \
-										 "<span class='notice'>You shave [H]'s facial hair clean off.</span>")
+										 "<span class='notice'>You shave [H]'s facial hair.</span>")
 					shave(H, location)
```
Comment (Shadowmech88) [inline code/game/objects/items/weapons/cosmetics.dm]: This needs to pass `user` in as an argument.

File: code/game/objects/items/weapons/cosmetics.dm [hunk 3]
```diff
@@ -332,8 +341,8 @@
 				user.visible_message("<span class='warning'>[user] tries to shave [H]'s head with [src]!</span>", \
 									 "<span class='notice'>You start shaving [H]'s head...</span>")
 				if(do_after(user, H, 50))
-					user.visible_message("<span class='warning'>[user] shaves [H]'s head bald with [src]!</span>", \
-										 "<span class='notice'>You shave [H]'s head bald.</span>")
+					user.visible_message("<span class='warning'>[user] shaves [H]'s head with [src]!</span>", \
+										 "<span class='notice'>You shave [H]'s head.</span>")
 					shave(H, location)
```
Comment (Shadowmech88) [inline code/game/objects/items/weapons/cosmetics.dm]: Same here.

**Comment by Durgen** [comment]
Italian speech disability when

**Comment by Durgen** [comment]
this job needs creates the imminent need for an Italian speech disability

**Comment by SonixApache** [comment]
Does the razor pop up a menu or can you switch modes on it with attack_self? The former would be really annoying for mass clowning.

File: code/game/jobs/access.dm [hunk 4]
```diff
@@ -68,6 +68,7 @@
 /var/const/access_weapons = 66 //Weapon authorization for secbots
 /var/const/access_taxi = 67 // Taxi drivers
 /var/const/access_shop = 68
+/var/const/access_barber = 69 //Barbers
```
Comment (ghost) [inline code/game/jobs/access.dm:71]: henk

File: code/game/objects/items/weapons/cosmetics.dm (hunk 2 above)
Comment (CptWad) [inline code/game/objects/items/weapons/cosmetics.dm]: so shave(H, user, location)?

**Comment by CptWad** [comment]
I have been informed that this might affect prefrences, not sure, may have to do a migration whatever that means
image

**Comment by CptWad** [comment]
wow it works better than I thought.
image
Some issues
- needs to prompt hairstyle before the timer
- because it does after, there's no next-to-person check meaning you can walk away and change their hair from a distance

**Comment by Riekland** [comment]
ATOMICITY REEEEEE

**Comment by CptWad** [comment]
only affects 1 map :^)

File: code/game/objects/items/weapons/cosmetics.dm [hunk 5]
```diff
@@ -275,17 +275,26 @@
 	icon_state = "razor"
 	w_class = W_CLASS_TINY
 	starting_materials = list(MAT_IRON = 340)
-
-/obj/item/weapon/razor/proc/shave(mob/living/carbon/human/H, location = "mouth")
+/obj/item/weapon/razor/proc/shave(mob/living/carbon/human/H, mob/user, location = "mouth")
 	if(location == "mouth")
```
Comment (Intigracy) [inline code/game/objects/items/weapons/cosmetics.dm]: Slap an adjacent check right here and you'll fix the problem you mentioned

Comment (Intigracy) [inline code/game/objects/items/weapons/cosmetics.dm:279]: Slap an adjacent check right here and you'll fix the second problem you mentioned

File: code/game/objects/items/weapons/cosmetics.dm [hunk 6]
```diff
@@ -294,47 +303,47 @@
...
 				to_chat(user,"<span class='warning'>Already clean-shaven!</span>")
-				return
+				return*/
 
 			if(H == user) //shaving yourself
-				user.visible_message("[user] starts to shave their facial hair with [src].", \
-									 "<span class='notice'>You take a moment to shave your facial hair with [src]...</span>")
+				user.visible_message("[user] starts to style their facial hair with [src].", \
+									 "<span class='notice'>You take a moment to style your facial hair with [src]...</span>")
 				if(do_after(user, H, 5))
```
Comment (Intigracy) [inline code/game/objects/items/weapons/cosmetics.dm:314]: Or put shave(H, user, location) here and handle the shaving in shave(), that way you can give them the prompt, then do the do_after after they've decided.

**Comment by CptWad** [comment]
Moving the adjacency didnt have the desired effect. I can still move away from people. I should add it again elsewhere

**Comment by CptWad** [comment]
man I hate coding but I also dont want to just let the barber go around and ONLY shave off peoples hair. this shits hard

File: code/game/objects/items/weapons/cosmetics.dm (hunk 5 above)
Comment (ghost) [reply code/game/objects/items/weapons/cosmetics.dm:279]: Is the adjacency check here?

**Comment by SonixApache** [comment]
i for one pity wad

**Pull request closed without merging**